### PR TITLE
Upgrade to Rails 6

### PR DESCRIPTION
**Pull request by ahorincar**
Upgrading the Rails version to Rails 6. Added necessary changes to the dummy app and removed unnecessary gem version dependencies. 

I changed the gem version to 3.1.0 (from 2.1.0) since this upgrade is a major upgrade for any host app using it (even though we know that atm it's just one app).

### CODE REVIEW

File: lib/helena/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module Helena
-  VERSION = '2.1.0'.freeze
+  VERSION = '3.1.0'.freeze
```
Comment (crackofdusk) [inline lib/helena/version.rb]: Wouldn't we bump to 3.0.0?

File: helena.gemspec [hunk 2]
```diff
@@ -19,20 +19,20 @@ survey/test development, longitudinal studies and instant feedback.'
...
+  s.add_dependency 'mongoid'
+  s.add_dependency 'mongoid-simple-tags'
+  s.add_dependency 'mongoid-tree'
+  s.add_dependency 'mongoid_orderable'
+  s.add_dependency 'rails', '~> 6.0.3.4'
+  s.add_dependency 'rails-i18n'
+  s.add_dependency 'responders'
+  s.add_dependency 'sass-rails'
+  s.add_dependency 'simple_form'
+  s.add_dependency 'slim'
```
Comment (crackofdusk) [inline helena.gemspec:31]: I'm not sure if dropping the constraints on the production dependencies is a good idea, but I also don't know which versions Helena really supports.

Comment (ahorincar) [reply helena.gemspec:31]: Well, they all have `>=`, so I think they were put there to guard against older versions

File: lib/helena/version.rb (hunk 1 above)
Comment (ahorincar) [reply lib/helena/version.rb]: Yup, good catch

**Review by crackofdusk — APPROVED** [review]

**Pull request merged**